sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQuerySourceBase.java

```diff
@@ -168,10 +170,12 @@ public void validate() {
 
     SerializableFunction<GenericRecord, TableRow> function =
         new SerializableFunction<GenericRecord, TableRow>() {
+          private Supplier<TableSchema> schema = new SerializableTableSchemaSupplier(
```

Comment:
Hmm, can this be simplified by making this a `Suppliers.memoize(Suppliers.compose(BigQueryHelpers::fromJsonString, Suppliers.ofInstance(jsonSchema)))` ? (except the lambda has to be Java 7)

using methods from http://google.github.io/guava/releases/22.0/api/docs/com/google/common/base/Suppliers.html